### DISCUSSION TITLE
open and close subinteraction overlay to start point

### DISCUSSION THREAD

**Discussion by hpvd** [unpoly/unpoly#534] [Ideas]
The "Branch off subinteractions into layers" is really nice!
see https://demo.unpoly.com/projects

Maybe one can optimize it further by giving the user some orientation where the new layer belongs to.
This could be done by using the position of the "parent" as start and end point of animation,
see
https://codepen.io/Sahil89/pen/ddzYjy

**Comment by hpvd** [comment]
if interested in, these are also helpful:

window can be changed, reduced in size or changed in direction (turn mobile/smartphone/tablet direction)

https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

`The trick is to apply a CSS transform to the modal in order to deform into the shape/position of the clicked note before showing the modal. We can then remove the transform and use a CSS transition to get it to smoothly animate into it's natural shape/position.`
https://stackoverflow.com/a/65056071

**Reply by alexandremjacques** [reply]
I guess it depends on your use case. But, you could try `up.layer.parent`:

https://unpoly.com/layer-option#matching-relative-to-the-current-layer

https://unpoly.com/up.layer.parent
